File: platform/packages_py/env_resolver/env_resolver/env_rally_x.py
from dataclasses import dataclass
from typing import Any

from env_resolve import resolve


@dataclass(frozen=True)
class RallyEnv:
    api_key: str | None
    base_url: str


def resolve_rally_env(config: dict[str, Any] | None = None) -> RallyEnv:
    return RallyEnv(
        api_key=resolve(None, ['RALLY_API_KEY'], config, 'api_key', None),
        base_url=resolve(None, ['RALLY_BASE_URL'], config, 'base_url', 'https://rally1.rallydev.com'),
    )

File: platform/packages_mjs/env-resolver/src/env-rally.x.ts
import { resolve } from '@internal/env-resolve';

export interface RallyEnv {
    apiKey: string | undefined;
    baseUrl: string;
}

export function resolveRallyEnv(config?: Record<string, any>): RallyEnv {
    return {
        apiKey: resolve(undefined, ['RALLY_API_KEY'], config, 'apiKey', undefined),
        baseUrl: resolve(undefined, ['RALLY_BASE_URL'], config, 'baseUrl', 'https://rally1.rallydev.com'),
    };
}
